dashdblocal_notebooks/livy-notebooks/IoT_Demo_Pyspark.ipynb

```diff
@@ -0,0 +1,1246 @@
+    "print(\"Showing Newest \"+str(df_check.count()) +\" Records:\")\n",
+    "df_check.select(\"deviceId\", \"time\", \"temperature\", \"tempOutside\", \"powerProd\", \"noiseLevel1\").\\\n",
+    "                orderBy(\"time\", ascending=False).show()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Visualisation of time series\n",
+    "\n",
+    "You could want to visualize the data you are collecting. Let's plot some of the features over time."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "__Note : Make a bridge between remote and local__\n",
+    "\n",
+    "So far we've been using Spark servers and the data is accessed remotely. If we plot visualisations now, they won't appear right on your screen : you will have to fetch them, using their adress, to open them locally. So for convenience, visualisations will be made locally thanks to the __%local__ magic. Therefore, we need to create a temporary view of the data, and query it with the %%sql magic in order to access it locally and plot it directly. Note that the data on the Spark cluster is a Spark dataframe, whereas the data we handle locally is a pandas dataframe.\n",
+    "\n",
+    "In the following cells, we will:\n",
+    "1. Create a SQL temporary view of the tornado (Spark) dataframe\n",
+    "2. Run a %%sql cell magic to query the data, and output it to the local space. The table is now seen as a Pandas dataframe.\n",
+    "3. We use this Pandas dataframe to make our graphs, which we output locally."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 42,
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "application/vnd.jupyter.widget-view+json": {
+       "model_id": "",
+       "version_major": 2,
+       "version_minor": 0
+      },
+      "text/plain": [
+       "FloatProgress(value=0.0, bar_style='info', description='Progress:', layout=Layout(height='25px', width='50%'),…"
+      ]
+     },
+     "metadata": {},
+     "output_type": "display_data"
+    }
+   ],
+   "source": [
+    "df_check.createOrReplaceTempView('turbine')"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 43,
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "application/vnd.jupyter.widget-view+json": {
+       "model_id": "",
+       "version_major": 2,
+       "version_minor": 0
+      },
+      "text/plain": [
+       "FloatProgress(value=0.0, bar_style='info', description='Progress:', layout=Layout(height='25px', width='50%'),…"
+      ]
+     },
+     "metadata": {},
+     "output_type": "display_data"
+    },
+    {
+     "name": "stderr",
+     "output_type": "stream",
+     "text": [
+      "/Users/eva.feillet/anaconda3/lib/python3.7/site-packages/autovizwidget/widget/utils.py:50: FutureWarning:\n",
+      "\n",
+      "A future version of pandas will default to `skipna=True`. To silence this warning, pass `skipna=True|False` explicitly.\n",
```

Comment:
Just to make the notebook a bit nicer, do what the warning says and it will disappear.

Reply:
yes I agree, I have already tried but in fact I can't pass `skipna=True|False`

Reply:
Just to clarify: are you trying to pass `skipna=True|False` literally? The `|` means "or", so you can pass either `skipna=True` or `skipna=False`, but not `skipna=True|False` :-)

Reply:
no not 'skipna=True|False' literally, I mean I get an error when I pass skipna=True to get read of the warning

Reply:
Ok, but which error is that?

Reply:
My bad, this is a `%%sql` magic cell, you can't fix that warning.